services/worker-manager/src/providers/azure.js

```diff
@@ -837,8 +871,16 @@ class AzureProvider extends Provider {
       // record operation (NOTE: this information is never used, as deletion is tracked
       // by name)
       await worker.modify(w => {
-        w.providerData[resourceType].id = false;
-        w.providerData[resourceType].operation = deleteRequest.getPollState().azureAsyncOperationHeaderValue;
+        let resource;
+        if (index !== undefined) {
+          resource = w.providerData[resourceType][index];
+        } else {
+          resource = w.providerData[resourceType];
+        }
+        resource.id = false;
+        if (_.has(deleteRequest, 'azureAsyncOperationHeaderValue')) {
```

Comment:
This is looking for a property of the `getPollState()` return value, but on `deleteRequest` itself.  I suspect this will never be true..